### PR TITLE
Add autoExport

### PR DESCRIPTION
**Pull request by timneutkens**
Implements https://github.com/zeit/next.js/issues/7355

This automatically statically exports all pages in Carbon.

It also adds the new version of `@now/next` which automatically adds immutable headers.

### CODE REVIEW

File: now.json [hunk 1]
```diff
@@ -22,13 +20,9 @@
         "cache-control": "public, max-age=43200, immutable"
       }
     },
+    { "handle": "filesystem" },
```
Comment (mfix22) [inline now.json]: @timneutkens what does this line do? I have seen it before, but never in the documentation?

**Comment by mfix22** [comment]
@timneutkens the build is failing with this error:
```
{ Error: Cannot find module '/home/travis/build/dawnlabs/carbon/.next/server/pages-manifest.json'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:690:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Server.generatePublicRoutes (/home/travis/build/dawnlabs/carbon/node_modules/next-server/dist/server/next-server.js:220:31)
    at Server.generateRoutes (/home/travis/build/dawnlabs/carbon/node_modules/next-server/dist/server/next-server.js:151:33)
    at new Server (/home/travis/build/dawnlabs/carbon/node_modules/next-server/dist/server/next-server.js:64:29)
    at module.exports (/home/travis/build/dawnlabs/carbon/node_modules/next-server/index.js:4:10)
    at module.exports.options (/home/travis/build/dawnlabs/carbon/node_modules/next/dist/server/next.js:2:161)
    at start (/home/travis/build/dawnlabs/carbon/node_modules/next/dist/server/lib/start-server.js:1:323)
    at nextStart (/home/travis/build/dawnlabs/carbon/node_modules/next/dist/cli/next-start.js:22:125)
    at commands.(anonymous function).then.exec (/home/travis/build/dawnlabs/carbon/node_modules/next/dist/bin/next:29:346)
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at Function.Module.runMain (internal/modules/cjs/loader.js:832:11)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3) code: 'MODULE_NOT_FOUND' }
```

**Comment by mfix22** [comment]
Also this is the styling issue we were noticing earlier when we tried to upgrade to `8.1`. Looks like it is still present: 
Screenshot 2019-06-09 16 34 03

File: now.json (hunk 1 above)
Comment (mfix22) [reply now.json]: @timneutkens @timothyis can one of you point me to the documentation for this^?

**Comment by mfix22** [comment]
> Also this is the styling issue we were noticing earlier when we tried to upgrade to `8.1`. Looks like it is still present:
> Screenshot 2019-06-09 16 34 03

@timneutkens we are still seeing the styling issues ^. These have been present since `8.1.0`.

File: now.json (hunk 1 above)
Comment (lfades) [reply now.json]: @mfix22 Here: https://zeit.co/docs/v2/deployments/routes#wildcard-routes

**Comment by ijjk** [comment]
Hi, @mfix22 the styling issue should be updated in the latest canary `v8.1.1-canary.53`

**Comment by timneutkens** [comment]
This was the fix: https://github.com/zeit/next.js/pull/7571

File: now.json (hunk 1 above)
Comment (mfix22) [reply now.json]: Does it matter what order this command is in the routes array though?

**Comment by mfix22** [comment]
@timneutkens @ijjk not sure how "autoExport" works so it will be hard for me to debug, but `next-offline` is now broken — no service worker was registered. 

> SW registration failed:  DOMException

**Comment by ijjk** [comment]
This isn't due to `autoExport` and is from how we handle the `public` folder now. You can update `nextOffline` to output the `service-worker` to the `public` folder. For example:

```js
const path = require('path')
const withOffline = require('next-offline')

module.exports = withOffline({
  dontAutoRegisterSw: true,
  workboxOpts: {
    swDest: path.join(__dirname, 'public/service-worker.js'),
  },
  target: 'serverless',
  experimental: {
    autoExport: true
  }
})
```

File: next.config.js [hunk 2]
```diff
@@ -1,3 +1,9 @@
 const withOffline = require('next-offline')
 
-module.exports = withOffline({ dontAutoRegisterSw: true })
+module.exports = withOffline({
+  target: 'serverless',
+  dontAutoRegisterSw: true,
+  experimental: {
```
Comment (ijjk) [inline next.config.js]:
```suggestion
  workboxOpts: {
    swDest: path.join(__dirname, 'public/service-worker.js'),
  },
  experimental: {
```

Comment (mfix22) [reply next.config.js]: @ijjk this it now working, but the exported `service-worker.js` looks incorrect. Were there breaking changes with how the files are exported? 

(https://carbon-9gvevwmgy.now.sh/service-worker.js)
```js
self.__precacheManifest = [
  {
    "url": "/_next/static/A6_zdG4b__6z6u6~Yvxr6/pages/_app.js",
    "revision": "baaf6a34d4e46f39ab1b"
  },
  {
    "url": "/_next/static/A6_zdG4b__6z6u6~Yvxr6/pages/_error.js",
    "revision": "82986cfd500b031b842d"
  },
  {
    "url": "/_next/static/A6_zdG4b__6z6u6~Yvxr6/pages/about.js",
    "revision": "7a52c3d6947ad4d6069a"
  },
  {
    "url": "/_next/static/A6_zdG4b__6z6u6~Yvxr6/pages/embed.js",
    "revision": "0a1aae8cef7c59539916"
  },
  {
    "url": "/_next/static/A6_zdG4b__6z6u6~Yvxr6/pages/index.js",
    "revision": "f74a6f915575adf6d224"
  },
  {
    "url": "/_next/static/chunks/4f3d.4f29a31c9b2405e7f1eb.js",
    "revision": "f2ec1d304f122dd7f441"
  },
  {
    "url": "/_next/static/chunks/7021.c9608c316ac8728ec42c.js",
    "revision": "24e31e618e3ec49d740e"
  },
  {
    "url": "/_next/static/chunks/8fda.208c4594c728152ed538.js",
    "revision": "883598af10579d079b50"
  },
  {
    "url": "/_next/static/chunks/cbfc.8ec4593a7acccfa77c95.js",
    "revision": "9edebaef2c40e9ff8f60"
  },
  {
    "url": "/_next/static/chunks/commons.c116e5e03b31e084745e.js",
    "revision": "3ca1ce41bfb359c73baf"
  },
  {
    "url": "/_next/static/chunks/ea35.67b67cb20915e22ef28c.js",
    "revision": "dd9b75a7b744b8f1bae9"
  },
  {
    "url": "/_next/static/runtime/main-d1b8e7abd637fc72076d.js",
    "revision": "3577e73338e258137e8c"
  },
  {
    "url": "/_next/static/runtime/webpack-0947b9a85b5e00afae80.js",
    "revision": "a7356b692cf50ee77e1c"
  }
];

/**
 * Welcome to your Workbox-powered service worker!
 *
 * You'll need to register this file in your web app and you should
 * disable HTTP caching for this file too.
 * See https://goo.gl/nhQhGp
 *
 * The rest of the code is auto-generated. Please don't update this file
 * directly; instead, make changes to your Workbox build configuration
 * and re-run your build process.
 * See https://goo.gl/2aRDsh
 */

importScripts("https://storage.googleapis.com/workbox-cdn/releases/4.3.1/workbox-sw.js");

importScripts(
  
);

self.addEventListener('message', (event) => {
  if (event.data && event.data.type === 'SKIP_WAITING') {
    self.skipWaiting();
  }
});

/**
 * The workboxSW.precacheAndRoute() method efficiently caches and responds to
 * requests for URLs in the manifest.
 * See https://goo.gl/S9QRab
 */
self.__precacheManifest = [].concat(self.__precacheManifest || []);
workbox.precaching.precacheAndRoute(self.__precacheManifest, {});
```

Comment (ijjk) [reply next.config.js]: Why does it look incorrect, are there files not there that used to be?

Comment (mfix22) [reply next.config.js]: @ijjk it's not that it is incorrect, but rather that the service worker is being registered, but failing when items are requested (see deployment)

Do we need to add this line back? https://github.com/dawnlabs/carbon/pull/782/files#diff-96c813af76efae3c872399365c060e54L9

File: now.json [hunk 3]
```diff
@@ -2,15 +2,14 @@
   "name": "carbon",
   "version": 2,
   "public": true,
-  "builds": [{ "src": "package.json", "use": "@now/static-build" }],
+  "builds": [{ "src": "package.json", "use": "@now/next@canary" }],
   "routes": [
     { "src": "^/about(.*)", "dest": "/about" },
     { "src": "^/embed(.*)", "dest": "/embed" },
     { "src": "^/_next/(.*)", "dest": "/_next/$1" },
```
Comment (mfix22) [inline now.json]: @timneutkens can I remove this line again now? Or is it still required?

Comment (timneutkens) [reply now.json]: Done!

**Comment by mfix22** [comment]
@timneutkens this is the only thing I am trying to get fixed:
Screenshot 2019-06-27 10 33 00

The app doesn't work in "offline" mode.

File: now.json (hunk 1 above)
Comment (timneutkens) [reply now.json]: Yes, it goes over every route 1 by 1

**Pull request merged**